recbole/trainer/trainer.py

```diff
@@ -577,8 +578,11 @@ def evaluate(
             return
 
         if load_best_model:
+            # Refer to: https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#save-and-load-checkpoints
+            dist.barrier()
             checkpoint_file = model_file or self.saved_model_file
-            checkpoint = torch.load(checkpoint_file, map_location=self.device)
+            map_location = {"cuda:%d" % 0: "cuda:%d" % self.config["local_rank"]}
+            checkpoint = torch.load(checkpoint_file, map_location=map_location)
```

Comment:
Fix the `EOFError: Ran out of input` error when using DDP.

Refer to the example in the [ddp tutorial](https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#save-and-load-checkpoints).